ground/gcs/src/plugins/uavtalk/telemetry.h

```diff
@@ -84,7 +84,9 @@ class Telemetry : public QObject
     Telemetry(UAVTalk *utalk, UAVObjectManager *objMngr);
     ~Telemetry();
     TelemetryStats getStats();
-    QByteArray *downloadFile(quint32 fileId, quint32 maxSize);
+    QByteArray *downloadFile(quint32 fileId, quint32 maxSize,
+            std::function<void(quint32)>progressCb = nullptr);
```

Comment:
I don't know if I agree, but if we had a fully-async version it can all be signals (whether same thread or different).

it's just when it's synchronous, it seems dumb to connect/disconnect.